src/ia-reviews.ts

```diff
@@ -0,0 +1,408 @@
+  }
+
+  /** Handles successful review submission */
+  private handleReviewUpdate(e: CustomEvent<Review>): void {
+    // Update the reviews count to reflect new review, if applicable
+    if (!this.currentReview) {
+      this.reviewsCount = this.reviewsCount + 1;
+    }
+
+    this.currentReview = e.detail;
+    this.displayReviewForm = false;
+  }
+
+  /** Closes the review form if requested */
+  private handleEditCanceled(): void {
+    this.displayReviewForm = false;
+
+    // Display no-reviews message again if no new review was added
+    if (this.reviewsCount === 0) {
+      this.displayReviews = false;
+    }
+  }
+
+  static get styles(): CSSResultGroup {
+    return [
+      iaButtonStyles,
+      css`
+        :host {
+          font-family: var(
+            --ia-font-stack,
+            'Helvetica Neue',
+            Helvetica,
+            Arial,
+            sans-serif
+          );
+
+          color: var(--ia-text-color, #2c2c2c);
+          display: flex;
+          flex-direction: row;
+          gap: 2rem;
+          align-items: flex-start;
+        }
+
+        .reviews-list {
+          display: flex;
+          flex-direction: column;
+          gap: 2rem;
+        }
+
+        .own-review-container {
+          --error-color: var(--container-error-color, #ea0202);
+          --link-color: var(--container-link-color, #4f65f5);
+          --container-error-color: #ea0202;
+          --container-link-color: #4f65f5;
+          --container-bg-color: #fbfbfd;
+          --container-border-color: #999999;
+
+          border: 2px solid var(--container-border-color, #999999);
+          border-radius: 5px;
+          background-color: var(--container-bg-color, #fbfbfd);
+          padding: 10px;
+        }
+
+        .message {
+          font-weight: 200;
+        }
+
+        .message .ia-button {
+          display: inline;
+          vertical-align: baseline;
+          padding: 0;
+          font-weight: 600;
+        }
+
+        .reviews-title {
+          display: flex;
+          flex-direction: row;
+          justify-content: space-between;
+          align-items: flex-end;
+          border-bottom: 1px solid #979797;
+          padding-bottom: 5px;
+          margin-bottom: 10px;
+        }
+
+        .reviews-title .reviews-icon {
+          vertical-align: bottom;
+          display: none;
+        }
+
+        .reviews-icon {
+          width: 3.5rem;
+          padding-top: 0.5rem;
+        }
+
+        .reviews {
+          flex-grow: 1;
+        }
+
+        .reviews-title h2 {
+          font-size: 3rem;
+          font-weight: 500;
+          margin: 0;
+        }
+
+        .add-edit-btn {
+          all: unset;
+          font-weight: 500;
+        }
+
+        .add-edit-btn:hover {
+          cursor: pointer;
+        }
+
+        .add-icon {
+          width: 1.6rem;
+          display: inline-block;
+          vertical-align: bottom;
+        }
+
+        @media only screen and (max-width: 767px) {
```

Comment:
So I think this is mostly fine (and not a blocker), but one thing I've tried to do when possible is, instead of changing the UI based on the screen size, change the UI based on its own available UI space using the `SharedResizeObserver`. 

If you take the use case where the reviews section isn't necessarily inside a full-screen-width section, this same logic may not work, whereas if you know that the icon should be displayed whenever the reviews section is at least, say 500px, that gives you much more finer-grained control over it and it will always be correct. We do something like this on the home page where we change the layout based on the width of the home page (which isn't always full-width):

https://git.archive.org/www/offshoot/-/blob/main/src/scenes/home/home-page.ts#L231-234

Reply:
Cool yeah I see that behavior! This works!

It's just a thing that pops into my head (it's a bit of a web component purist mindset) but also not something to fully stick to. 

I guess to expand on that thought process, it makes me think of "what is the page layout responsible for vs what the web component responsible for"? Like I could see the page being responsible for providing the header line for the "Reviews" section since we could potentially have other sections on the page that have header lines like that and the component do all the functional work inside that section. 

Just thinking out loud here. Nothing to change :)